mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -524,18 +548,26 @@ def create_model_version(
         headers, run = self._get_run_and_headers(run_id)
         source_workspace_id = self._get_workspace_id(headers)
         notebook_id = self._get_notebook_id(run)
+        table_detail_list = self._get_table_details(run)
```

Comment:
Out of curiosity is there any limit on how long this list can be (how many MLflow datasets in a run)? Just so we don't pass a huge header in the request if there are too many tables. We could truncate if there's no limit

Reply:
No limit on datasets per run today. Not sure how this backend service works, but I presume you have some server-side protection against large requests, which might be good enough here.

Reply:
I'll limit it to 10 for now.  We can revisit in the final CR.